pkg/services/extsvcauth/oauthserver/oasimpl/service.go

```diff
@@ -134,44 +137,51 @@ func (s *OAuth2ServiceImpl) GetExternalService(ctx context.Context, id string) (
 		return nil, err
 	}
 
-	// Handle the case where the external service has no service account
+	if err := s.setClientUser(ctx, client); err != nil {
+		return nil, err
+	}
+
+	s.cache.Set(id, *client, cacheExpirationTime)
+	return client, nil
+}
+
+// setClientUser sets the SignedInUser and SelfPermissions fields of the client
+func (s *OAuth2ServiceImpl) setClientUser(ctx context.Context, client *oauthserver.OAuthExternalService) error {
 	if client.ServiceAccountID == oauthserver.NoServiceAccountID {
-		s.logger.Debug("GetExternalService: service has no service account, hence no permission", "id", id, "name", client.Name)
-		// Create a signed in user with no role and no permissions
+		s.logger.Debug("GetExternalService: service has no service account, hence no permission", "client_id", client.ClientID, "name", client.Name)
+
+		// Create a signed in user with no role and no permission
 		client.SignedInUser = &user.SignedInUser{
 			UserID:      oauthserver.NoServiceAccountID,
 			OrgID:       oauthserver.TmpOrgID,
 			Name:        client.Name,
 			Permissions: map[int64]map[string][]string{oauthserver.TmpOrgID: {}},
 		}
-		s.cache.Set(id, *client, cacheExpirationTime)
-		return client, nil
+		s.cache.Set(client.ClientID, *client, cacheExpirationTime)
```

Comment:
I guess this line is redundant, because https://github.com/grafana/grafana/pull/77248/files#diff-43716239c7f3517490eb419b2e18612225345e35a574f382a9ccb9e79f690ef7R144 is handling the caching.